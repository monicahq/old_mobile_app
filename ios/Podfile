# Uncomment the next line to define a global platform for your project
# platform :ios, '9.0'

target 'Monica' do
  pod 'AppCenter/Crashes', '~> 1.9.0'
  pod 'AppCenter/Analytics', '~> 1.9.0'
  pod 'AppCenterReactNativeShared', '~> 1.8.0'
  # Uncomment the next line if you're using Swift or would like to use dynamic frameworks
  # use_frameworks!

  # Pods for Monica


  platform :ios, '9.0'
  target 'MonicaTests' do
    inherit! :search_paths
    # Pods for testing
  end

end

target 'Monica-tvOS' do
  # Uncomment the next line if you're using Swift or would like to use dynamic frameworks
  # use_frameworks!

  # Pods for Monica-tvOS

  target 'Monica-tvOSTests' do
    inherit! :search_paths
    # Pods for testing
  end

end
