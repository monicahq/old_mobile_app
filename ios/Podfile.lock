PODS:
  - AppCenter/Analytics (1.9.0):
    - AppCenter/Core
  - AppCenter/Core (1.9.0)
  - AppCenter/Crashes (1.9.0):
    - AppCenter/Core
  - AppCenterReactNativeShared (1.8.0):
    - AppCenter/Core (= 1.9.0)

DEPENDENCIES:
  - AppCenter/Analytics (~> 1.9.0)
  - AppCenter/Crashes (~> 1.9.0)
  - AppCenterReactNativeShared (~> 1.8.0)

SPEC REPOS:
  https://github.com/cocoapods/specs.git:
    - AppCenter
    - AppCenterReactNativeShared

SPEC CHECKSUMS:
  AppCenter: 3cf7b9387549e9703dab068793e404013e8c0cc1
  AppCenterReactNativeShared: 53eb73a85b3adaf0d5b6dd8ea276a56e2d9aed6c

PODFILE CHECKSUM: b0c19c98bd5484d6dd08724f52840eee715339ef

COCOAPODS: 1.5.3
