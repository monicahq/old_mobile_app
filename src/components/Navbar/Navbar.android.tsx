import React, {PureComponent} from 'react';
import Icon from 'react-native-vector-icons/MaterialIcons';

import {NetworkStatusContainer} from '@containers/NetworkStatus';
import {commonStyles} from '@theme';
import {INavbarProps} from './Navbar.props';

const actions = [
  {title: 'Close', iconName: 'close', iconSize: 30, show: 'always'},
];

export class Navbar extends PureComponent<INavbarProps> {
  public onActionSelected = () => {
    this.props.onBack();
  };

  public render() {
    const {onBack, title} = this.props;

    return [
      <Icon.ToolbarAndroid
        key={0}
        logo={require('@assets/logo.png')}
        title={'  ' + title}
        titleColor="white"
        onActionSelected={this.onActionSelected}
        style={commonStyles.toolbarAndroid}
        actions={onBack && actions}
      />,
      <NetworkStatusContainer key={1} />,
    ];
  }
}
