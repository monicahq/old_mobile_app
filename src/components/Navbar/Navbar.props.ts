import {ReactNode} from 'react';

export interface INavbarProps {
  title: string | ReactNode;
  onBack?: any;
  rightTitle?: string;
  rightAction?: (...args) => void;
  rightIcon?: string;
}
