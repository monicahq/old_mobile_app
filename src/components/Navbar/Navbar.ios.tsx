import React, {PureComponent} from 'react';
import {TouchableOpacity} from 'react-native';
import NavigationBar from 'react-native-navbar';
import Icon from 'react-native-vector-icons/Ionicons';

import {hitSlop, navbarColor, statusBarLightContent} from '@theme';
import {INavbarProps} from './Navbar.props';
import {styles} from './Navbar.styles';

export class Navbar extends PureComponent<INavbarProps> {
  public render() {
    const {onBack, title} = this.props;

    return (
      <NavigationBar
        style={styles.navbar}
        rightButton={this.getRightButton()}
        leftButton={
          onBack && (
            <TouchableOpacity
              onPress={onBack}
              style={styles.leftButton}
              hitSlop={hitSlop}
            >
              <Icon name="ios-arrow-back" size={30} color="white" />
            </TouchableOpacity>
          )
        }
        title={typeof title === 'string' ? {title, tintColor: 'white'} : title}
        tintColor={navbarColor}
        statusBar={statusBarLightContent}
      />
    );
  }
  private getRightButton() {
    const {rightAction, rightTitle} = this.props;
    if (!rightTitle) {
      return null;
    }
    return {
      title: rightTitle,
      handler: rightAction,
      tintColor: 'white',
    };
  }
}
