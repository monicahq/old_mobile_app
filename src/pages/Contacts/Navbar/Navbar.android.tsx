import React, {PureComponent} from 'react';
import SearchBar from 'react-native-searchbar';
import Icon from 'react-native-vector-icons/MaterialIcons';

import {NetworkStatusContainer} from '@containers/NetworkStatus';
import {I18n} from '@i18n';
import {commonStyles} from '@theme';
import {INavbarProps} from './Navbar.props';

const actions = [
  {
    title: I18n.t('common:search'),
    iconName: 'search',
    iconSize: 30,
    show: 'always',
  },
];

export class Navbar extends PureComponent<INavbarProps, {}> {
  private searchBar;

  public onActionSelected = () => {
    this.searchBar.show();
  };
  public onBack = () => {
    this.searchBar.hide();
    this.props.onSearchTextChanged('');
  };
  public hide = () => {
    this.searchBar.hide();
  };
  public searchBarRef = ref => {
    this.searchBar = ref;
  };
  public render() {
    return [
      <Icon.ToolbarAndroid
        logo={require('@assets/logo.png')}
        key={0}
        title={'  ' + I18n.t('contacts:contacts')}
        titleColor="white"
        onActionSelected={this.onActionSelected}
        style={commonStyles.toolbarAndroid}
        actions={actions}
      />,
      <SearchBar
        key={1}
        ref={this.searchBarRef}
        placeholder={I18n.t('contacts:search')}
        onBlur={this.hide}
        handleSearch={this.props.onSearchTextChanged}
        onBack={this.onBack}
      />,
      <NetworkStatusContainer key={2} />,
    ];
  }
}
