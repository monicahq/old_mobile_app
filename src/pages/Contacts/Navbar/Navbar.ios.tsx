import React, {PureComponent} from 'react';
import {Image} from 'react-native';
import NavigationBar from 'react-native-navbar';
import SearchBar from 'react-native-search-bar';

import {I18n} from '@i18n';

import {NetworkStatusContainer} from '@containers/NetworkStatus';
import {navbarColor, statusBarLightContent} from '@theme';
import {INavbarProps} from './Navbar.props';
import {styles} from './Navbar.styles';

export class Navbar extends PureComponent<INavbarProps, {}> {
  public render() {
    return [
      <NavigationBar
        key={0}
        leftButton={
          <Image
            source={require('@assets/logo.png')}
            style={styles.navbarLogo}
          />
        }
        title={
          <SearchBar
            placeholder={I18n.t('contacts:search')}
            hideBackground={true}
            showsCancelButtonWhileEditing={false}
            autoCapitalize="words"
            searchBarStyle="minimal"
            textColor="#ACACAC"
            onChangeText={this.props.onSearchTextChanged}
            style={styles.searchBar}
          />
        }
        rightButton={<NetworkStatusContainer />}
        tintColor={navbarColor}
        statusBar={statusBarLightContent}
      />,
    ];
  }
}
