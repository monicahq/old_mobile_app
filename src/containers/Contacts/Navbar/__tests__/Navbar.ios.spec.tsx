import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import {Navbar} from '../Navbar.ios';

jest.mock('@navigator/NavigationService', () => ({
  navigate: jest.fn(),
}));

describe('Components', () => {
  describe('Navbar Contacts IOS', () => {
    const onSearchTextChanged = jest.fn();
    // const contact = {first_name: 'plouf', last_name: 'nathie'};

    it('should renders correctly', () => {
      expect(
        toJson(shallow(<Navbar onSearchTextChanged={onSearchTextChanged} />))
      ).toMatchSnapshot();
    });

    it('should renders correctly with onBack', () => {
      expect(
        toJson(
          shallow(
            <Navbar
              onSearchTextChanged={onSearchTextChanged}
              onBack={jest.fn()}
            />
          )
        )
      ).toMatchSnapshot();
    });
  });
});
