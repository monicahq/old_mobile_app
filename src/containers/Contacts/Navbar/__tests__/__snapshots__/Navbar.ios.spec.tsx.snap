// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Navbar Contacts IOS should renders correctly 1`] = `
<NavigationBar
  containerStyle={Object {}}
  leftButton={
    <Component
      source={1}
      style={
        Object {
          "marginLeft": 8,
          "marginTop": 7,
        }
      }
    />
  }
  rightButton={null}
  statusBar={
    Object {
      "style": "light-content",
    }
  }
  style={
    Object {
      "marginTop": 0,
    }
  }
  tintColor="#325776"
  title={
    <SearchBar
      autoCapitalize="words"
      autoCorrect={false}
      barStyle="default"
      barTintColor={null}
      cancelButtonText="Cancel"
      editable={true}
      enablesReturnKeyAutomatically={true}
      hideBackground={true}
      keyboardAppearance="default"
      keyboardType="default"
      onBlur={[Function]}
      onCancelButtonPress={[Function]}
      onChange={[Function]}
      onChangeText={[MockFunction]}
      onFocus={[Function]}
      onSearchButtonPress={[Function]}
      placeholder="Search your contacts"
      returnKeyType="search"
      searchBarStyle="minimal"
      showsCancelButton={false}
      showsCancelButtonWhileEditing={false}
      spellCheck={false}
      style={
        Object {
          "height": 35,
          "marginTop": 4,
          "width": 245,
        }
      }
      text=""
      textColor="#ACACAC"
      textFieldBackgroundColor={null}
      tintColor={null}
    />
  }
/>
`;

exports[`Components Navbar Contacts IOS should renders correctly with onBack 1`] = `
<NavigationBar
  containerStyle={Object {}}
  leftButton={
    <NavbarIOSBack
      onPress={[MockFunction]}
    />
  }
  rightButton={null}
  statusBar={
    Object {
      "style": "light-content",
    }
  }
  style={
    Object {
      "marginTop": 0,
    }
  }
  tintColor="#325776"
  title={
    <SearchBar
      autoCapitalize="words"
      autoCorrect={false}
      barStyle="default"
      barTintColor={null}
      cancelButtonText="Cancel"
      editable={true}
      enablesReturnKeyAutomatically={true}
      hideBackground={true}
      keyboardAppearance="default"
      keyboardType="default"
      onBlur={[Function]}
      onCancelButtonPress={[Function]}
      onChange={[Function]}
      onChangeText={[MockFunction]}
      onFocus={[Function]}
      onSearchButtonPress={[Function]}
      placeholder="Search your contacts"
      returnKeyType="search"
      searchBarStyle="minimal"
      showsCancelButton={false}
      showsCancelButtonWhileEditing={false}
      spellCheck={false}
      style={
        Object {
          "height": 35,
          "marginTop": 4,
          "width": 245,
        }
      }
      text=""
      textColor="#ACACAC"
      textFieldBackgroundColor={null}
      tintColor={null}
    />
  }
/>
`;
